libs/dal/src/repositories/tenant/tenant.entity.ts

```diff
@@ -0,0 +1,22 @@
+import { TenantCustomData } from '@novu/shared';
+import { TenantId } from './types';
+import { EnvironmentId } from '../environment';
+import { ChangePropsValueType } from '../../types/helpers';
+
+export class TenantEntity {
+  _id: TenantId;
+
+  identifier: string;
+
+  name?: string;
+
+  deleted?: boolean;
+
+  createdAt: string;
+
+  data?: TenantCustomData;
+
+  _environmentId: EnvironmentId;
+}
+
```

Comment:
Not sure if an organization id needed here, and if this entity is one to many with an organization.
meaning one tenant related to two organizations.

raised a question on the linear task as well.

Reply:
I think the hierarchy should be:
organisation > environment > tenant > workflow
so I think it could be added without harm.
But not sure if @scopsy and @LetItRock could think in a different way.

Reply:
Thinking about it again now i think we can add the organizationId as well, and not mix the tenants across organizations by one to many relation as i said before 😅





> wouldn't this be a tenant per env? I feel like we should have here just the `_organizationId` and no `_environmentId`, wdyt?

i am not sure because we would want to separate the tenant entity on the environment level so for example `production` level won't be affected by the `test` environment therefore more secure in a way.